### PR TITLE
Update confusion_matrix.py

### PR DESCRIPTION
**Pull request by wanghonglie**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Update confusion_matrix.py

## Modification
Update confusion_matrix.py

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: tools/analysis_tools/confusion_matrix.py [hunk 1]
```diff
@@ -75,27 +76,26 @@ def calculate_confusion_matrix(dataset,
...
         else:
-            res_bboxes = per_img_res
-        ann = dataset.get_ann_info(idx)
-        gt_bboxes = ann['bboxes']
-        labels = ann['labels']
-        analyze_per_img_dets(confusion_matrix, gt_bboxes, labels, res_bboxes,
-                             score_thr, tp_iou_thr, nms_iou_thr)
+            res_bboxes = per_img_res['pred_instances']
+        gts = dataset.get_data_info(idx)['instances']
+        # gt_bboxes = ann['bboxes']
```
Comment (RangiLyu) [inline tools/analysis_tools/confusion_matrix.py:89]: clean the code

File: tools/analysis_tools/confusion_matrix.py [hunk 2]
```diff
@@ -242,28 +254,19 @@ def main():
         cfg.merge_from_dict(args.cfg_options)
 
     results = mmcv.load(args.prediction_path)
```
Comment (RangiLyu) [inline tools/analysis_tools/confusion_matrix.py:256]:
```suggestion
    results = mmengine.fileio.load(args.prediction_path)
```

File: tools/analysis_tools/confusion_matrix.py [hunk 3]
```diff
@@ -8,9 +8,10 @@
 from mmcv import Config, DictAction
```
Comment (RangiLyu) [inline tools/analysis_tools/confusion_matrix.py:8]:
```suggestion
from mmengine import Config, DictAction
```

**Review by RangiLyu — DISMISSED** [review]

File: tools/analysis_tools/confusion_matrix.py [hunk 4]
```diff
@@ -75,27 +77,24 @@ def calculate_confusion_matrix(dataset,
         tp_iou_thr (float|optional): IoU threshold to be considered as matched.
             Default: 0.5.
     """
-    num_classes = len(dataset.CLASSES)
+    num_classes = len(dataset.metainfo['CLASSES'])
     confusion_matrix = np.zeros(shape=[num_classes + 1, num_classes + 1])
     assert len(dataset) == len(results)
     prog_bar = mmcv.ProgressBar(len(results))
     for idx, per_img_res in enumerate(results):
         if isinstance(per_img_res, tuple):
```
Comment (ZwwWayne) [inline tools/analysis_tools/confusion_matrix.py:85]: there will be no such a case when the prediction results is a tuple, we should clean the logic.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**